docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,101 @@
+= Git Submodule guideline
+
+This document outlines some basic guidelines for synchronizing changes between an upstream and downstream repository using Git submodules.
+
+By leveraging Git submodules, it is possible to maintain a consistent and up-to-date relationship between related repositories (upstream and downstream) while having more control of how the project is built, tested and released.
+
+Additionally, the integration with Konflux can automate key aspects of this process, ensuring that changes from the upstream repository are efficiently and reliably propagated to the downstream repository.
+
+== Implementing this guideline
+=== Basic definitions
+
+- **Upstream repo**: a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+- **Downstream repo**: a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+=== Create a downstream repo
+
+The downstream repo can be used to do the following:
+
+* hold the git submodule reference to the upstream repository.
+* use the contents of the git submodule in a Containerfile for further edits.
+* onboarded onto {ProductName}.
+
+==== Configuring components
+
+CAUTION: Make sure to properly set the xref:/how-tos/creating.doc#context-and-containerfile-path[Context and Containerfile path] as required for the artifact builds.
```

Comment:
Since I might do this instead, just add the caution without the xref. You can, however, suggest what context might be needed (i.e. keep it at the repo root so that you can have the Contaienrfile and the submodule within the context).

Reply:
Sounds good, the changes have been applied here: https://github.com/konflux-ci/docs/pull/120/files#diff-8168286ae4fbd44cc1f9e5677aa3a5949ae97f5f391b1466cc253efe6176ccf8R26-R27